android/capacitor/src/main/java/com/getcapacitor/util/InternalUtils.java

```diff
@@ -0,0 +1,22 @@
+package com.getcapacitor.util;
+
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.os.Build;
+
+public class InternalUtils {
+
+    public static PackageInfo getPackageInfo(PackageManager pm, String packageName) throws PackageManager.NameNotFoundException {
+        return InternalUtils.getPackageInfo(pm, packageName, 0);
+    }
+
+    @SuppressWarnings("deprecation")
+    public static PackageInfo getPackageInfo(PackageManager pm, String packageName, long flags)
+        throws PackageManager.NameNotFoundException {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            return pm.getPackageInfo(packageName, PackageManager.PackageInfoFlags.of(flags));
+        } else {
+            return pm.getPackageInfo(packageName, (int) flags);
```

Comment:
can you move the deprecated code to a separate private function with the suppress warning and remove the suppress warning from the public one?
so in case google deprecates the new function in the future we don't miss it